server/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -299,7 +304,14 @@ public String toString() {
 
     @Override
     public RestStatus status() {
-        return isTimedOut() ? RestStatus.REQUEST_TIMEOUT : RestStatus.OK;
+        if (RestApiVersion.current() == RestApiVersion.V_7 && isTimedOut()) {
+            deprecationLogger.critical(DeprecationCategory.API, "cluster_health_request_timeout",
+                "HTTP status code for an internal cluster health timeout will change from 408 to 200 in a future major version");
+        }
+        if (Boolean.parseBoolean(System.getProperty("es.cluster_health.request_timeout_408", "false"))) {
+            return RestStatus.REQUEST_TIMEOUT;
+        }
+        return RestStatus.OK;
```

Comment:
If the user requested v7 compatibility and we timed out then we should keep the old behaviour. That would mean there's no need to skip any tests in v7-compatible mode.

Reply:
Is there a way to set a system property for the REST compatibility tests?